aces/Loris
# [document_repository] New features

## Brief summary of changes

- Changed view permission to only allow viewing of documents (Only browse tab accessible)
- Added an upload and edit permission (allows users to upload a document or to edit any documents in doc repo)
- Created a new permission: hidden files. Allows users that have the permission to hide files from users that do not have it.
- New tab: "Edit Categories". Allows to change the name of a category as well as its parent category.
- Created a new permission: categories. The edit categories and add categories do not appear unless you have this permission.

#### Testing instructions (if applicable)

1. Check with another account (not admin) that the permissions listed above behave as expected (the tabs appear as expected).
2. Upload a document when you have the hidden file permission and set the file to restricted. Check with an account that does not have the hidden file permission that this file does not appear in the doc tree. Check that it appears when you have the permission. Also check that when uploading a file or editing one, the restrict access option is not available if you do not have the hidden file permission
3. With the categories permission, create a new permission. Check that the category has the right name under the right parent category. Using the edit categories tab, edit the name of the category and change the parent category. Check that the category was moved accordingly and has the right name.
4. When editing a file, you should be able to change the category of the file.

#### Link(s) to related issue(s)

* Resolves #7099

@LeighMac please let me know if these features were the ones you were looking for.
